plugins/aws/lambda/envVarsClientSideEncryption.js

```diff
@@ -0,0 +1,90 @@
+    title: 'Lambda Environment Variables Client Side Encryption',
+    category: 'Lambda',
+    description: 'Ensure that all sensitive AWS Lambda environment variable values are client side encrypted.',
+    more_info: 'AWS Lambda lets you encrypt environment variable values prior to sending them to Lambda. ' +
+        'Environment variables are often used to store sensitve information such as passwords. Such variable valuesshould be ' +
+        'encrypted for security best practices.',
+    link: 'https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html',
+    recommended_action: 'Encrypt environment variables that store sensitive information',
+    apis: ['Lambda:listFunctions'],
+    settings: {
+        lambda_sensitive_env_vars: {
+            name: 'Lambda Sensitive Environment Varibales',
+            description: 'A comma-delimited list of known lambda function Environment Varibales that should be encrypted',
+            regex: '^.{1,255}$',
+            default: 'password'
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var config = {
+            lambda_sensitive_env_vars: settings.lambda_sensitive_env_vars || this.settings.lambda_sensitive_env_vars.default
+        };
+
+        if (!config.lambda_sensitive_env_vars.length) return callback(null, results, source);
+
+        config.lambda_sensitive_env_vars = config.lambda_sensitive_env_vars.split(',');
+
+        async.each(regions.lambda, function(region, rcb){
+            var listFunctions = helpers.addSource(cache, source,
+                ['lambda', 'listFunctions', region]);
+
+            if (!listFunctions) return rcb();
+
+            if (listFunctions.err || !listFunctions.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query for Lambda functions: ${helpers.addError(listFunctions)}`, region);
+                return rcb();
+            }
+
+            if (!listFunctions.data.length) {
+                helpers.addResult(results, 0, 'No Lambda functions found', region);
+                return rcb();
+            }
+
+            for (var lambdaFunc of listFunctions.data) {
+                if (!lambdaFunc.FunctionArn) continue;
+                var resource = lambdaFunc.FunctionArn;
+
+                if (!lambdaFunc.Environment || !lambdaFunc.Environment.Variables) {
+                    helpers.addResult(results, 0,
+                        'No environment variables found', region, resource);
+                    continue;
+                }
+
+                var unencryptedVars = [];
+                var encryptedVars = [];
+                for (var envVar of config.lambda_sensitive_env_vars) {
+                    if (lambdaFunc.Environment.Variables[envVar]) {
+                        if (lambdaFunc.Environment.Variables[envVar].length < 200) unencryptedVars.push(envVar);
```

Comment:
is it always the case that an encrypted var will have length > 200. what if we put ssh private key or some long string whose length is 200 and that is not encrypted. is there any other way to detect encryption.

Reply:
So Gio and I discussed and there is no way to know whether it is cipher text or value. So I changed the check to check for length to be exactly equal to 216 bytes as it is always that long.